modules/benchmark/R/calc.metrics.R

```diff
@@ -12,44 +11,74 @@
   
-  # Right now many argument are being read in as R list objects
-  # This won't work remotely but at the moment I don't have a good solution
+  # Right now I'm making the inappropriate assumption that storage type will be 
+  # empty unless it's a time variable. 
+  # This is because I haven't come up for a good way to test that a character string is a date format
+  st <- format$vars$storage_type
   
-  # create results table
-  results <- as.data.frame(matrix(NA, nrow = length(vars_used$pecan_name), ncol = length(metrics$name)))
-  rownames(results) <- vars_used$pecan_name
-  colnames(results) <- metrics$name
+  time.row <- which(nchar(st)>1 & substr(st, 1,1) == "%")
   
-  # Do I need these?
-  # start.year <- lubridate::year(settings$run$start.date)
-  # end.year <- lubridate::year(settings$run$end.date)
+  # ---- MODEL DATA ---- #
   
-  model <- as.data.frame(read.output(runid=basename(model_run), outdir=model_run, start.year=start_year, end.year=end_year, format$vars$pecan_name))
-  # returns model output list object with variable values (including time), variable name and units
+  model_vars <- format$vars$pecan_name[-time.row]
+  model <- as.data.frame(read.output(runid=basename(model_run), outdir=model_run, 
+                                     start.year=start_year, end.year=end_year, 
+                                     c("time",model_vars)))
+  vars.used.index <- which(format$vars$pecan_name %in% names(model)[!names(model)=="time"])
   
-  vars_used <- as.data.frame(matrix(NA, nrow = length(names(model)), ncol = ncol(format$vars)))
-  colnames(vars_used) <- colnames(format$vars)
-  for(i in  1:length(names(model))){
-    vars_used[i,] <- format$vars[which(format$vars$pecan_name == names(model)[i]),]
+  # We know that the model output time is days since the beginning of the year.
+  # Make a column of years to supplement the column of days of the year.
+  years <- start_year:end_year
+  Diff <- diff(model$time)
+  n <- model$time[c(which(Diff < 0), length(model$time))]
+  y <- c()
+  for(i in 1:length(n)){
+    y <- c(y, rep(years[i], n[i]))
   }
+  model$year <- y
+  model$posix <- strptime(paste(model$time, model$year), format = "%j %Y")
+  model_full = model
   
-  obvs <- load.data(data.path, format, start_year, end_year, site=settings$run$site, vars_used)
-  # observation list object with variable values (including time), variable name and units
+  # ---- INPUT DATA ---- #
   
-  dat <- align.data(model, obvs, vars_used, start_year, end_year)
+  obvs <- load.data(data.path, format, start_year, end_year, site, vars.used.index, time.row)
+  dat_vars <- format$vars$pecan_name[vars.used.index]
+  obvs_full = obvs
   
-  for(v in 1:length(vars_used$pecan_name)){
-    metric_dat <- dat[,c(paste(vars_used$pecan_name[v], c("model", "obvs"),sep = "_" ),"time")]
+  #########################
+  
+  source("modules/benchmark/R/mean.over.larger.timestep.R")
```

Comment:
1) this will fail when compiled as a package
2) this should be unnecessary when compiled as a package so long as the function is exported

I'd encourage you to make liberal use of RStudio's 'build' tab and to set the directory to the package you're working on. I've been doing this lately and rebuilding the packages I'm working on constantly, and I find this has made development much simpler. It can even handle calling roxygen for you.